src/main/java/games/strategy/engine/lobby/server/db/UserController.java

```diff
@@ -60,37 +71,42 @@ public boolean doesUserExist(final String userName) {
 
   @Override
   public void updateUser(final DBUser user, final HashedPassword hashedPassword) {
+    updateUser(user, hashedPassword, hashedPassword.isBcrypted() ? "bcrypt_password" : "password");
+  }
+
+  private void updateUser(final DBUser user, final HashedPassword hashedPassword, final String column) {
     Preconditions.checkArgument(user.isValid(), user.getValidationErrorMessage());
 
     try (final Connection con = connectionSupplier.get();
         final PreparedStatement ps =
-            con.prepareStatement("update ta_users set password = ?,  email = ?, admin = ? where username = ?")) {
+            con.prepareStatement("update ta_users set " + column + " = ?,  email = ? where username = ?")) {
       ps.setString(1, hashedPassword.value);
       ps.setString(2, user.getEmail());
-      ps.setInt(3, user.isAdmin() ? 1 : 0);
-      ps.setString(4, user.getName());
+      ps.setString(3, user.getName());
       ps.execute();
       con.commit();
     } catch (final SQLException e) {
-      throw new IllegalStateException(
-          "Error updating name:" + user.getName() + " email: " + user.getEmail() + " pwd: " + hashedPassword.mask(), e);
+      throw new IllegalStateException(String.format("Error updating name: %s email: %s pwd: %s",
+          user.getName(), user.getEmail(), hashedPassword.mask()), e);
     }
   }
 
   @Override
   public void createUser(final DBUser user, final HashedPassword hashedPassword) {
+    createUser(user, hashedPassword, hashedPassword.isBcrypted() ? "bcrypt_password" : "password");
+  }
+
+  private void createUser(final DBUser user, final HashedPassword hashedPassword, final String passwordColumn) {
     Preconditions.checkState(hashedPassword.isValidSyntax());
     Preconditions.checkState(user.isValid(), user.getValidationErrorMessage());
 
     try (final Connection con = connectionSupplier.get();
         final PreparedStatement ps = con.prepareStatement(
-            "insert into ta_users (username, password, email, joined, lastLogin, admin) values (?, ?, ?, ?, ?, ?)")) {
+            "insert into ta_users (username, " + passwordColumn + ", email, admin) values (?, ?, ?, ?)")) {
       ps.setString(1, user.getName());
       ps.setString(2, hashedPassword.value);
       ps.setString(3, user.getEmail());
-      ps.setTimestamp(4, new Timestamp(System.currentTimeMillis()));
-      ps.setTimestamp(5, new Timestamp(System.currentTimeMillis()));
-      ps.setInt(6, user.isAdmin() ? 1 : 0);
+      ps.setInt(4, user.isAdmin() ? 1 : 0);
```

Comment:
My original intent was to remove this call, but apparently it was used by tests only.
I kept it because we really should implement a way to change privileges from inside the engine.

Reply:
If this is used by tests only, should we even keep it at all?